### ISSUE TITLE
Bug: [member-ordering] natural-sort should be implemented using String's native capabilities instead of a 3rd party package

### ISSUE THREAD

**Issue by julienw** [typescript-eslint/typescript-eslint#5989]
I left a message on the PR that implemented this rule => https://github.com/typescript-eslint/typescript-eslint/pull/5662#discussion_r1021616113
But I thought it was better to file a new issue instead.

I saw that #5662 implemented a natural-sort-ordering option, which sounds good. However this can be implemented using node's native API instead of relying on a 3rd-party lib.

Would you be open to a PR for this?

**Comment by JoshuaKGoldberg** [comment]
Interesting! Which native API are you referring to?

**Comment by julienw** [comment]
ooops, sorry, I should have mentioned it here. I mentioned it in the message that I linked :)

I'm refering to the native JS API `String.prototype.localeCompare` that node supports for a long time. I tried a node version as old as v10 locally. Here is a very simple code example:
```js
["a2", "a10", "a1"].sort((a, b) => a.localeCompare(b, undefined, { numeric: true }))
// Result: [ 'a1', 'a2', 'a10' ]
```

Hope this helps

**Comment by JoshuaKGoldberg** [comment]
Ah thanks - so this actually isn't Node-specific, it's JS strings in general. That threw me - we support running in browsers and therefore try not to rely on Node-specific APIs.

I tried the suggested change out locally and saw a couple of incorrect errors. For example:

```plaintext
Member a5 should be declared before member a10.
```

The `natural-compare-lite` package compares groups of numbers against each other numerically, not alphabetically. That's not the same as using `localCompare`. 

Am I missing something?

**Comment by julienw** [comment]
I'd like to see your code :) Especially have you used `{numeric: true}` as an option? _That's_ the secret sauce.

**Comment by julienw** [comment]
My understandng:
natural-compare-lite orders with upper case letters first
localeCompare orders with lower case letters first

Normally we can configure this in `localeCompare` using `caseFirst: "upper"` but... I couldn't make it work (not in node, not in Firefox).

We can also decide that localeCompare order is the right one -- this is just a choice after all.

I asked around to colleagues about the issue though.

**Comment by JoshuaKGoldberg** [comment]
Sounds good! We can always [look at an issue for new sort options](https://github.com/typescript-eslint/typescript-eslint/issues/new?assignees=&labels=enhancement%3A+plugin+rule+option%2Cpackage%3A+eslint-plugin%2Ctriage&template=02-enhancement-rule-option.yaml&title=Enhancement%3A+%5Brule-name%5D+%3Ca+short+description+of+my+proposal%3E) if you feel there's a variant that should be added.

**Comment by julienw** [comment]
I got the bottom of this.

1. `localeCompare` sorts lexicographically. This means that `A` and `a` always come before `B` and `b`.
2. the `caseFirst` option configures whether `A` comes before or after `a`. The default is locale-dependent.

Therefore with `localeCompare` we can't have both the numeric sorting and have all uppercases come before lowercases like the 3rd-party library does.
